invenio_vocabularies/resources/config.py

```diff
@@ -0,0 +1,108 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2024 CERN.
+# Copyright (C) 2024 University of Münster.
+#
+# Invenio-Vocabularies is free software; you can redistringibute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""Resources config."""
+
+import marshmallow as ma
+from flask_resources import (
+    HTTPJSONException,
+    ResourceConfig,
+    create_error_handler,
+    JSONSerializer,
+    BaseListSchema,
+    MarshmallowSerializer,
+    ResponseHandler,
+)
+from invenio_records_resources.resources.records.headers import etag_headers
+from invenio_records_resources.resources.errors import ErrorHandlersMixin
+from invenio_records_resources.resources.records.args import SearchRequestArgsSchema
+from invenio_records_resources.services.base.config import ConfiguratorMixin
+from invenio_records_resources.resources import (
+    RecordResource,
+    RecordResourceConfig,
+    SearchRequestArgsSchema,
+)
+from .serializer import VocabularyL10NItemSchema
+
+
+class VocabularySearchRequestArgsSchema(SearchRequestArgsSchema):
+    """Vocabularies search request parameters."""
+
+    tags = ma.fields.Str()
+    active = ma.fields.Boolean()
+    status = ma.fields.Boolean()
+
+
+class VocabulariesResourceConfig(RecordResourceConfig):
+    """Vocabulary resource configuration."""
+
+    blueprint_name = "vocabularies"
+    url_prefix = "/vocabularies"
+    routes = {
+        "list": "/<type>",
+        "item": "/<type>/<pid_value>",
+        "tasks": "/tasks",
+    }
+
+    request_view_args = {
+        "pid_value": ma.fields.Str(),
+        "type": ma.fields.Str(required=True),
+    }
+
+    request_search_args = VocabularySearchRequestArgsSchema
+
+    response_handlers = {
+        "application/json": ResponseHandler(JSONSerializer(), headers=etag_headers),
+        "application/vnd.inveniordm.v1+json": ResponseHandler(
+            MarshmallowSerializer(
+                format_serializer_cls=JSONSerializer,
+                object_schema_cls=VocabularyL10NItemSchema,
+                list_schema_cls=BaseListSchema,
+            ),
+            headers=etag_headers,
+        ),
+    }
+
+
+class VocabularyTypeResourceConfig(ResourceConfig, ConfiguratorMixin):
+    """Vocabulary list resource config."""
+
+    # /vocabulary/vocabulary_id
+    # Blueprint configuration
+    blueprint_name = "vocabulary_list"
+    url_prefix = "/vocabularies"
+
+    routes = {
+        "all": "/",
+        "list": "/vocabularies/<pid_value>",
+        "item": "/vocabularies/<pid_value>/<vocabulary_type_id>",
+    }
```

Comment:
minor: won't this produce a `/vocabularies/vocabularies/<pid_value>` endpoint in the end? Though I could see that we're ending up with a name clash with the existing `/vocabularies/<pid_type>` endpoint (e.g. `/vocabularies/licenses`). Maybe we can hardcode having paths like `/vocabularies/types/<type_id>`?